crates/agglayer-interop-grpc-types/src/compat/v1/aggchain_data/types_to_v1.rs

```diff
@@ -0,0 +1,164 @@
+        Ok(v1::AggchainProof {
+            proof: Some(value.proof.try_into()?),
+            aggchain_params: Some(value.aggchain_params.into()),
+            signature: None,
+            context: match value.public_values {
+                Some(public_values) => HashMap::from([(
+                    "public_values".to_owned(),
+                    Bytes::from(
+                        std::panic::catch_unwind(|| bincode::serialize(&*public_values))
+                            .map_err(|_| {
+                                Error::serializing_aggchain_proof_public_values(Box::new(
+                                    bincode::ErrorKind::Custom(String::from("panic")),
+                                ))
+                            })?
+                            .map_err(Error::serializing_context)?,
+                    ),
+                )]),
+                None => HashMap::new(),
+            },
+        })
+    }
+}
+
+impl TryFrom<Proof> for v1::aggchain_proof::Proof {
+    type Error = Error;
+
+    fn try_from(value: Proof) -> Result<Self, Self::Error> {
+        match value {
+            Proof::SP1Stark(SP1StarkWithContext {
+                proof,
+                vkey,
+                version,
+            }) => Ok(v1::aggchain_proof::Proof::Sp1Stark(v1::Sp1StarkProof {
+                proof: std::panic::catch_unwind(|| sp1v4_bincode_options().serialize(&proof))
+                    .map_err(|_| {
+                        Error::serializing_proof(Box::new(bincode::ErrorKind::Custom(
+                            String::from("panic"),
+                        )))
+                    })?
+                    .map_err(Error::serializing_context)?
+                    .into(),
+                vkey: std::panic::catch_unwind(|| sp1v4_bincode_options().serialize(&vkey))
+                    .map_err(|_| {
+                        Error::serializing_vkey(Box::new(bincode::ErrorKind::Custom(String::from(
+                            "panic",
+                        ))))
+                    })?
+                    .map_err(Error::serializing_context)?
+                    .into(),
+                version,
+            })),
+        }
+    }
+}
+
+impl TryFrom<AggchainData> for v1::AggchainData {
+    type Error = Error;
+
+    fn try_from(value: AggchainData) -> Result<Self, Self::Error> {
+        Ok(v1::AggchainData {
+            data: Some(match value {
+                AggchainData::ECDSA { signature } => {
+                    v1::aggchain_data::Data::Signature(v1::FixedBytes65 {
+                        value: Bytes::copy_from_slice(&signature.as_bytes()),
+                    })
+                }
+                AggchainData::Generic {
+                    proof: Proof::SP1Stark(proof),
+                    signature,
+                    aggchain_params,
+                    public_values,
+                } => v1::aggchain_data::Data::Generic(v1::AggchainProof {
```

Comment:
Nit: There is some opportunity for further refactoring, making the conversion methods smaller. For example, the whole block starting on this line could be factored out as `TryFrom<AggchainProof> for v1::AggchainProof`. I am fairly certain there are more opportunities like this.